src/nvim/strings.c

```diff
@@ -45,6 +45,41 @@
 #include "nvim/os/os.h"
 #include "nvim/os/shell.h"
 
+/// Compares two strings for equality.
+bool strequal(const char *a, const char *b)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strcmp(a, b) == 0;
+}
+
+/// Compares two sized strings for equality.
+bool strnequal(const char *a, const char *b, size_t size)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strncmp(a, b, size) == 0;
+}
+
+/// Compares two strings for equality ignoring differences in case.
+bool striequal(const char *a, const char *b)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strcasecmp(a, b) == 0;
```

Comment:
So:

`strcasecmp`: modern POSIX function, not supported by Windows
`stricmp`: Windows function that appears to be neither POSIX nor ANSI, despite what some blog posts say, it is [deprecated by Microsoft](https://msdn.microsoft.com/en-us/library/ms235365.aspx). Strangely enough, even Microsoft refers to these functions as being POSIX.
`_stricmp`: Windows function that's apparently the non-deprecated version of `strcmp`.

Btw, [this SO post](http://stackoverflow.com/questions/12413585/difference-of-stricmp-and-stricmp-in-visual-studio) claims that `stricmp` is merely an alias to `stricmp` in MSVC, which wouldn't surprise me.

I woudl go for `ifdef`ing towards the underscore version though, since it's explicitly referenced as non-dreprecated.